library: Add Email Entry

SoNiC-HeRE opened the pull request:
Closes #329

SoNiC-HeRE commented:
@andyholmes  Can you please provide an early review for the email demo?
As of last time it seemed to work for everyone else, so just wanna make sure if it's a permission error or something else

sonnyp commented:
Please make sure to use `Gio._promisify`

SoNiC-HeRE commented:
> Please make sure to use `Gio._promisify` 

Yes , i was just making sure if it was a problem on my side

sonnyp commented on src/Library/demos/Email/main.js:
```diff
@@ -0,0 +1,36 @@
+      ["sonicworks05@gmail.com"],
+      ["test@gmail.com"],
+      null,
+      "Demo Message",
```
can you add a comment next to each line with the name of the argument so it's clear what is what?

For example here

```suggestion
      "Demo Message", // subject
```

sonnyp commented on src/Library/demos/Email/main.js:
```diff
@@ -0,0 +1,36 @@
+  try {
+    const result = await portal.compose_email(
+      parent,
+      ["sonicworks05@gmail.com"],
+      ["test@gmail.com"],
```
Please don't use your personal data in examples.

```suggestion
      ["example@example.com"], // addresses
      null, // cc
```

sonnyp replied:
Please don't use your personal data in examples.

```suggestion
      ["example@example.com"], // addresses
      null, // cc
```

sonnyp commented on src/Library/demos/Email/main.js:
```diff
@@ -0,0 +1,36 @@
+  }
+}
+
+button.connect("clicked", sendEmail);
```
I have updated our practice a bit here see https://github.com/sonnyp/Workbench/pull/339

Please remove the try catch from sendEmail to make the function simpler and do the following instead

```suggestion
button.connect("clicked", () => {
  onClicked().catch(logError);
));
```

sonnyp commented on src/Library/demos/Email/main.js:
```diff
@@ -0,0 +1,36 @@
+      ["test@gmail.com"],
+      null,
+      "Demo Message",
+      "Demo Content",
```
I would suggest "Email from Workbench" as subject and "Hello World!" as body

sonnyp commented on src/Library/demos/Email/main.json:
```diff
@@ -0,0 +1,7 @@
+{
+  "name": "Email",
+  "category": "platform",
+  "description": "Compose an email message",
```
remember to write description from the perspective of the developer

```suggestion
  "description": "Trigger an email message",
```

sonnyp reviewed (commented):
It would be nice to add a text input inviting the user to enter their email address and use it for the `addresses` arg.

This way, when the user can test by sending the email to themselves directly. 

wdyt?

sonnyp commented on src/Library/demos/Email/main.blp:
```diff
@@ -0,0 +1,25 @@
+
+Adw.StatusPage {
+  title: "Email";
+  description: _("Compose an email");
```
```suggestion
  description: _("Trigger an email");
```

SoNiC-HeRE commented:
> It would be nice to add a text input inviting the user to enter their email address and use it for the `addresses` arg.
> 
> This way, when the user can test by sending the email to themselves directly.
> 
> wdyt?

Sure, it's a good idea

SoNiC-HeRE commented:
Made the following changes
- Used a Gtk.Entry to take email from user
- Commented various properties of portal.compose_email method
- Implemented an email validator
- Refined Code
- Removed async handlers

SoNiC-HeRE commented:
Made the following changes
- Used a Gtk.Entry to take email from user
- Commented various properties of portal.compose_email method
- Implemented an email validator
- Refined Code
- Removed async handlers

sonnyp reviewed (approved):
Well done.

I made minor changes https://github.com/sonnyp/Workbench/pull/330/commits/a3966fbab420ef092b9d60dada7e211750842147

Make sure to use the name of arguments as defined by the API documentation next time.
